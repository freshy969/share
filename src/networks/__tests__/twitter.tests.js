import Twitter from '../twitter';

describe('Share > Twitter', () => {
  test('url - all parameters', () => {
    const twitter = Twitter.create({});
    Object.defineProperty(twitter, 'entityLink', {value: jest.fn(), writable: true});
    twitter.entityLink.mockReturnValueOnce('https://demo.frontity.com/the-beauties-of-gullfoss');

    expect(
      twitter.url({
        type: 'post',
        id: 60,
        text: 'The Beauties of Gullfoss',
        via: 'frontity.com',
        hashtags: ['frontity', 'gullfoss', 'beauties'],
      }),
    ).toMatchSnapshot();
  });

  test('url - required parameters', () => {
    const twitter = Twitter.create({});
    Object.defineProperty(twitter, 'entityLink', {value: jest.fn(), writable: true});
    twitter.entityLink.mockReturnValueOnce('https://demo.frontity.com/the-beauties-of-gullfoss');

    expect(
      twitter.url({
        type: 'post',
        id: 60,
      }),
    ).toMatchSnapshot();
  });
});
