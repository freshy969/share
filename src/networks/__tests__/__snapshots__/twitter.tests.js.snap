// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share > Twitter url 1`] = `"https://twitter.com/share?url=https%3A%2F%2Fdemo.frontity.com%2Fthe-beauties-of-gullfoss&text=The%20Beauties%20of%20Gullfoss&via=frontity.com&hashtags=frontity%2Cgullfoss%2Cbeauties"`;
